tools/luthier.luau

```diff
@@ -0,0 +1,373 @@
+	end
+end
+
+local function getProjectPath(): string
+	return projectRelative(getBuildDir())
+end
+
+local function projectPathExists(): boolean
+	local projectPath = getProjectPath()
+
+	return safeFsType(projectPath) == "dir"
+end
+
+local function getExePath(): string
+	local target = args:get("target")
+	local exeName = getExeName(target)
+
+	return projectRelative(getBuildDir(), exeName)
+end
+
+local function exeExists(): boolean
+	local exePath = getExePath()
+
+	return safeFsType(exePath) == "file"
+end
+
+local function call(cmd: { string }, passedCwd: string?): number
+	print("> " .. table.concat(cmd, " "))
+
+	local res = process.create(cmd, { cwd = passedCwd or cwd })
+	print(res.stdout)
+	if res.stderr ~= "" then
+		print(res.stderr)
+	end
+	return res.exitcode
+end
+
+local function build()
+	local targetName = args:get("target")
+	local projectPath = getProjectPath()
+
+	if isWindows then
+		targetName ..= ".exe"
+	end
+
+	if args:has("clean") then
+		call({ "ninja", "-C", projectPath, "clean" })
+	end
+
+	local cmd = { "ninja", "-C", projectPath, targetName }
+
+	return call(cmd)
+end
+
+local function getConfigureArguments()
+	local projectPath = getProjectPath()
+
+	local config = string.lower(getConfig())
+	if config == "debug" then
+		config = "Debug"
+	elseif config == "release" then
+		config = "Release"
+	end
+
+	if args:has("clean") then
+		call({ "rm", "-rf", projectPath })
+	end
+
+	local configArgs =
+		{ "-G=Ninja", "-B " .. projectPath, "-DCMAKE_BUILD_TYPE=" .. config, "-DCMAKE_EXPORT_COMPILE_COMMANDS=1" }
+
+	if args:get("cxx-compiler") then
+		table.insert(configArgs, "-DCMAKE_CXX_COMPILER=" .. args:get("cxx-compiler"))
+	end
+
+	if args:get("c-compiler") then
+		table.insert(configArgs, "-DCMAKE_C_COMPILER=" .. args:get("c-compiler"))
+	end
+
+	return configArgs
+end
+
+local function readTuneFile(path: string): Tune
+	return toml.deserialize(fs.readasync(path))
+end
+
+local function check(exitCode: number)
+	if exitCode ~= 0 then
+		error("Command failed with exit code " .. exitCode)
+	end
+end
+
+local function fetchDependency(dependencyInfo: Tune)
+	local dependency = dependencyInfo.dependency
+
+	if not dependency then
+		error("tune file does not have a dependency table")
+	end
+	if not dependency.remote or not dependency.branch then
+		error("Invalid dependency info in tune file: must have `remote` and `branch` fields")
+	end
+
+	if safeFsType(projectRelative("extern", dependency.name)) == "dir" then
+		check(
+			call(
+				{ "git", "fetch", "--depth=1", "origin", dependency.revision },
+				projectRelative("extern", dependency.name)
+			)
+		)
+		local result = call({ "git", "checkout", dependency.revision }, projectRelative("extern", dependency.name))
+		check(
+			call(
+				{ "git", "submodule", "update", "--init", "--recursive", "--depth=1" },
+				projectRelative("extern", dependency.name)
+			)
+		)
+		return result
+	end
+
+	if gitVersionInfo.major >= 2 and gitVersionInfo.minor >= 49 then
```

Comment:
This is buggy logic, it actually doesn't work when version is 3 or higher. The correct logic is:
```suggestion
	if gitVersionInfo.major >= 3 or (gitVersionInfo.major == 2 and gitVersionInfo.minor >= 49) then
```

Reply:
good catch, thanks